### DISCUSSION TITLE
[WebAPI!] Can I get image which is preprocessor generated, not only result through webui api?

### DISCUSSION THREAD

**Discussion by log901108** [Mikubill/sd-webui-controlnet#1807] [Q&A] [answered]
Hello, I am trying to using sd webui by 'web API' which is running on 7861 port not gui on web 7860.

I am struggling to get image which is controlnet preprocessors generated, however there is no arg to get it. Is there any way to get those image from preprocessor? web gui gives it through enable preview, however there is no way with api 7861.

Also anyone give some clues for saving intermidiate results image between other extensions like adetailer or ddetailer through api?

Thank you for any help!

**Comment by huchenlei** [comment, marked as answer]
You can use "controlnet/detect" endpoint to get preprocessor result.

**Reply by log901108** [reply]
Thank you very much! I'll try it!
